kotlin-runtime/src/main/kotlin/io/micronaut/context/env/hocon/HoconPropertySourceLoader.kt

```diff
@@ -36,22 +36,57 @@ import kotlin.collections.LinkedHashMap
  * A PropertySourceLoader that supports config4k.
  *
  * @author graemerocher
+ * @author Marcelo Liberato
  * @since 1.1
  */
 class HoconPropertySourceLoader : PropertySourceLoader, Ordered {
 
+    private val impl by lazy {
+        ClassUtils
+                .forName("io.micronaut.context.env.hocon.HoconPropertySourceLoaderImpl", HoconPropertySourceLoader::class.java.classLoader)
+                .get()
+                .newInstance() as PropertySourceLoader
+    }
+
     override fun getExtensions(): MutableSet<String> {
         return Collections.singleton("conf")
     }
-    
-    override fun load(resourceName: String?, resourceLoader: ResourceLoader?): Optional<PropertySource> {
-        return loadEnv(resourceName, resourceLoader, null)
-    }
 
     override fun isEnabled(): Boolean {
         return ClassUtils.isPresent("com.typesafe.config.Config", HoconPropertySourceLoader::class.java.classLoader)
     }
 
+    override fun load(resourceName: String?, resourceLoader: ResourceLoader?): Optional<PropertySource> {
+        return if(isEnabled) {
```

Comment:
Shouldn't this be `isEnabled()`?

Reply:
isEnabled (without parentheses) is in Kotlin property access format, suggested by IDE